lib/news_scraper/uri_parser.rb

```diff
@@ -0,0 +1,15 @@
+module NewsScraper
+  class URIParser
```

Comment:
Why not just use Ruby's `URI`?
`host` => `URI.host` for example, also why do we remove the scheme for the HTTP request? Half the servers will just redirect